docs/Time_in_Arbitrum.md

```diff
@@ -4,52 +4,55 @@ title: Time in Arbitrum
+A rollup chain can configure the maximum amount its clock can differ, but generally it will stay within 20 blocks or 10 minutes
+In particular, a rollup chain can configure how far behind it's clock may fall, but generally it will stay within 20 blocks or 10 minutes.
 
-As you would expect, the block time given by `block.number` will never decrease; it can only increase.
+As you would expect, the results given by `block.number` and `block.timestamp` will never decrease; they can only increase.
 
-If you want to know when the current message (the one that requested the transaction you're currently running) was timestamped, you can get it in Solidity by writing:
+If you want to know when the current message (the one that requested the transaction you're currently running) arrived, you can get it by the following Soldiity calls:
 
-    ArbSys(address(100)).currentMessageTime()
+    ArbSys(address(100)).currentMessageBlock()
+    ArbSys(address(100)).currentMessageTimestamp()
 
-At any time, you can get an upper bound on the true Ethereum block time by using this Solidity call:
+At any time, you can get an upper bound on the true Ethereum block number or timestamp by using these Solidity calls:
 
-    ArbSys(address(100)).timeUpperBound()
+    ArbSys(address(100)).blockUpperBound()
+    ArbSys(address(100)).timestampUpperBound()
 
-This will never be more than 20 blocks ahead of `block.number`.
-The real Ethereum block number will always be between the ArbChain's `block.number` and what timeUpperBound() would return (inclusive).
+This will have the same accurancy guarantee from the ArbChain's configuration described about. For example the `blockUpperBound()` can be configured to never be more than 20 blocks ahead of the current Ethereum block number.
+The real Ethereum block number and timestamp will always be between the ArbChain's `block.number`/`block.timestamp` and what `blockUpperBound()`/`timestampUpperBound()` would return (inclusive).
 
 Bear in mind that the ArbChain might "freeze" at any time, so the upper bound is only a real upper bound at the moment that you make that call.
 After the call, the Ethereum time might have advanced.
 
 ## Use Cases
 
-Most developers will be fine using `block.number`.
-If you don't mind a little bit of time lag, you can keep on using `block.number` just like you would on Ethereum.
+Most developers will be fine using `block.number` and `block.timestamp`.
+If you don't mind a little bit of time lag, you can keep on using `block.number` and `block.timestamp` just like you would on Ethereum.
 
 ### Setting deadlines for user response
 
 An exception is a use case like an auction, where the contract emits an event, and you want to give people out there in userland N block times to respond.
 To do that, you'll want to "start the clock ticking" by using code like this:
 
-    deadline = ArbSys(address(100)).timeUpperBound() + N;
+    deadline = ArbSys(address(100)).blockUpperBound() + N;
```

Comment:
Do we want to use timestamps rather than block numbers in this example?

Reply:
I think this makes sense. Along with that I changed it to `N hours` rather than `N blocks`